activesupport/html_in_i18n.md

```diff
@@ -0,0 +1,37 @@
+# Using HTML blocks in internationalisation (i18n) locale files
+
+In projects which make good use of locale files to provide the textual content of HTML pages, the need often arises for keys to include HTML elements as well as plain text. However, by default i18n string values are marked as unsafe for HTML usage, so any `<tag>` elements get escaped and display on screen.
+
+To get round this, it is tempting to use Rails' `.html_safe` method to manually clear a string for HTML usage and to avoid it being escaped:
+
+```haml
+# views/page.html.haml
+.tagline
+  = t('header.tagline', subject: subject).html_safe
+```
+
+```yml
+# locales/en.yml
+header:
+  tagline: "The best website for <strong>%{subject}</strong>"
+```
+
+However, this has the potential to allow `subject` to include rogue HTML (or worse, Javascript XSS vulnerabilities) which could then be rendered in the browser.
+
+An alternative is to name the key with an `_html` suffix. This allows us to use HTML elements within the text of our YAML locale file, but any parameters that are supplied are escaped. So if any of the data we are passing in has been supplied by users, they should still be safe to display.
+
+```haml
+# views/page2.html
```

Comment:
should this be `# views/page2.html.haml`? 
Also, would it make sense to put the examples in the same order as before? First the `yml` and then the `html`?